### PR TITLE
[MI-2101] Handled OutputImage response from VA

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
Uploaded the image sent by VA to mattermost using mattermost UploadFile function

### CODE REVIEW

File: server/plugin/virtual_agent.go [hunk 1]
```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
 		}
 	}
 
 	return nil
 }
 
+func (p *Plugin) CreateOutputImagePost(body *OutputImage, userID string) (*model.Post, error) {
+	channel, appErr := p.API.GetDirectChannel(userID, p.botUserID)
+	if appErr != nil {
+		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", userID, "error", appErr.Message)
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:288]:
```suggestion
		p.API.LogInfo("Couldn't get bot's DM channel", "UserID", userID, "Error", appErr.Message)
```
Why `LogInfo` here?
And are we using snake_case everywhere for keys in the logs?

File: server/plugin/virtual_agent.go [hunk 2]
```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+		return nil, appErr
+	}
+
+	resp, err := http.Get(body.Value)
+	if err != nil {
+		p.API.LogInfo("Error in getting file data from link", "error", err.Error())
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:294]:
```suggestion
		p.API.LogInfo("Error in getting file data from link", "Error", err.Error())
```
Do this everywhere.
Why using `LogInfo` everywhere?

File: server/plugin/virtual_agent.go [hunk 3]
```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+
+	linkContents := strings.Split(body.Value, "/")
+
+	if len(linkContents) < 1 {
+		p.API.LogInfo(InvalidImageLinkError)
+		return nil, errors.New(InvalidImageLinkError)
+	}
+
+	completeFilename := linkContents[len(linkContents)-1]
+
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:313]:
```suggestion
```

File: server/plugin/virtual_agent.go [hunk 4]
```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+	filenameContents := strings.Split(completeFilename, ".")
+
+	ContentTypeInHeaders := ""
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:316]:
```suggestion
	contentTypeInHeaders := ""
```

File: server/plugin/virtual_agent.go [hunk 5]
```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
...
+
+		filename := filenameContents[0]
+		completeFilename = fmt.Sprintf("%s.%s", filename, fileExtension)
+	}
+
+	post := &model.Post{}
+	file, appErr := p.API.UploadFile(data, channel.Id, completeFilename)
+	if appErr != nil {
+		post.Message = body.AltText
+		p.API.LogInfo("Couldn't upload file on mattermost", "channel_id", channel.Id, "error", appErr.Message)
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:340]:
```suggestion
		p.API.LogInfo("Couldn't upload file on Mattermost", "ChannelID", channel.Id, "Error", appErr.Message)
```

File: server/plugin/virtual_agent.go (hunk 4 above)
Comment (raghavaggarwal2308) [reply server/plugin/virtual_agent.go:316]: Fixed [here](https://github.com/Brightscout/mattermost-plugin-servicenow-virtual-agent-private/pull/6/files#diff-0f5f6f38127a9ea36ffecf95673cc3105903094fb27fa91e0774291963b3ddf9)

Comment (manojmalik20) [reply server/plugin/virtual_agent.go:373]: You shared a link of the private repo, this change should also be in the public repo.

File: server/plugin/virtual_agent.go [hunk 6]
```diff
@@ -395,7 +394,7 @@ func (p *Plugin) CreateOutputImagePost(body *OutputImage, userID string) (*model
 	file, appErr := p.API.UploadFile(data, channel.Id, completeFilename)
 	if appErr != nil {
 		post.Message = body.AltText
-		p.API.LogInfo("Couldn't upload file on mattermost", "channel_id", channel.Id, "error", appErr.Message)
+		p.API.LogError("Couldn't upload file on mattermost", "channelID", channel.Id, "Error", appErr.Message)
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:397]:
```suggestion
		p.API.LogError("Couldn't upload file on mattermost", "ChannelID", channel.Id, "Error", appErr.Message)
```

File: server/plugin/virtual_agent.go [hunk 7]
```diff
@@ -343,34 +344,32 @@ func (p *Plugin) ProcessResponse(data []byte) error {
 func (p *Plugin) CreateOutputImagePost(body *OutputImage, userID string) (*model.Post, error) {
 	channel, appErr := p.API.GetDirectChannel(userID, p.botUserID)
 	if appErr != nil {
-		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", userID, "error", appErr.Message)
+		p.API.LogError("Couldn't get bot's DM channel", "userID", userID, "Error", appErr.Message)
```
Comment (manojmalik20) [inline server/plugin/virtual_agent.go:347]:
```suggestion
		p.API.LogError("Couldn't get bot's DM channel", "UserID", userID, "Error", appErr.Message)
```

File: server/plugin/bot.go [hunk 8]
```diff
@@ -38,14 +38,14 @@ func (p *Plugin) DMWithAttachments(mattermostUserID string, attachments ...*mode
 func (p *Plugin) dm(mattermostUserID string, post *model.Post) (string, error) {
 	channel, err := p.API.GetDirectChannel(mattermostUserID, p.botUserID)
 	if err != nil {
-		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", mattermostUserID, "error", err.Error())
+		p.API.LogError("Couldn't get bot's DM channel", "userID", mattermostUserID, "Error", err.Error())
```
Comment (manojmalik20) [inline server/plugin/bot.go:41]:
```suggestion
		p.API.LogError("Couldn't get bot's DM channel", "UserID", mattermostUserID, "Error", err.Error())
```
Fix this everywhere.

**Review by manojmalik20 — APPROVED** [review]

File: server/plugin/virtual_agent.go (hunk 4 above)
Comment (raghavaggarwal2308) [reply server/plugin/virtual_agent.go:373]: Yeah its in the public repo as well

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**